charms/metallb-speaker/Pipfile.lock

```diff
@@ -0,0 +1,526 @@
+{
```

Comment:
It comes from [pipenv](https://pipenv.pypa.io/en/latest/) which is the [recommended dependency manager](https://packaging.python.org/guides/tool-recommendations/) now.  It's [generally recommended](https://pipenv.pypa.io/en/latest/basics/#general-recommendations-version-control) to keep the `Pipfile.lock` in version control, unless you're targeting multiple versions of Python; with `charmcraft`, I think the Python version should be locked at build time (via the venv), but that may need to be tested.  For K8s charms, however, the Python version is set by the Juju unit operator image, anyway.

Reply:
Hrm.  It seems that I was mistaken and [`charmcraft` doesn't bundle or lock the Python version](https://discourse.juju.is/t/what-files-would-you-expect-charmcraft-build-copies-into-the-charm/3247/12).  I still think that including the `Pipfile.lock` in version control makes more sense for our use-case, though, since I think the consensus is that updating dependencies should be an explicit step rather than something that happens at every build.  (The reactive charm build process follows the latter approach and it has required charm authors to jump through quite a few hoops to work around that to ensure the ability to rebuild a previous release to add patches, for example.)